src/vm/vm_memory/memory.rs

```diff
@@ -100,6 +100,20 @@ impl Memory {
         }
     }
 
+    //Gets the value from memory address given by a MaybeRelocatable.
+    //If the value is an MaybeRelocatable::Int(Bigint) return &Bigint
+    //else raises Err
+    pub fn get_integer_from_maybe_relocatable(
+        &self,
+        key: &MaybeRelocatable,
+    ) -> Result<&BigInt, VirtualMachineError> {
+        match self.get(key) {
+            Ok(Some(MaybeRelocatable::Int(int))) => Ok(int),
+            Ok(_) => Err(VirtualMachineError::ExpectedInteger(key.clone())),
+            Err(memory_error) => Err(VirtualMachineError::MemoryError(memory_error)),
+        }
+    }
+
```

Comment:
It's probably better to just implement an `as_int` method for for `MaybeRelocatable`.
Also, let's just start using `Relocatable` for addresses. We can make `Memory::get` generic to allow for both temporarily I think.

Reply:
After the previous refactors this function is no longer needed, and can be replaced with get_integer. I think making Memory::get generic is a good idea, but it exceeds the scope of this PR